lib/payment_processor/braintree/webhook_handler.rb

```diff
@@ -17,16 +17,31 @@ class WebhookHandler
       # * +:notification+    - Braintree::Notification object. only those of kind 'subscription_charged_successfully'
       #                        will be processed. All others will simply be logged to the Rails logger.l)
 
-      def self.handle(notification)
-        new(notification).handle
+      def self.handle(signature, payload)
+        new(signature, payload).handle
       end
 
-      def initialize(notification)
-        @notification = notification
+      def initialize(signature, payload)
+        @signature = signature
+        @payload = payload
       end
 
       def handle
-        case @notification.kind
+        store_notification
+        process_notification
+      end
+
+      private
+
+      def store_notification
+        Payment::Braintree::Notification.create(
+          signature: @signature,
+          payload:   @payload
+        )
+      end
+
+      def process_notification
+        case notification.kind
         when 'subscription_charged_successfully'
           handle_subscription_charged
         when 'subscription_canceled'
```

Comment:
Do we need to push anything to the queue in this case?
I see you're using a timestamp for the cancelled field. I made booleans for GoCardless. I guess I should change that.